core/src/main/java/org/apache/accumulo/core/client/admin/compaction/CompactionConfigurer.java

```diff
@@ -31,6 +31,13 @@
  * @since 2.1.0
  */
 public interface CompactionConfigurer {
+
+  /**
+   * Property that can be set on the table, or provided by the override method below for this
+   * compaction. When true, we will call setDropBehind on the majc output files.
+   */
+  String TABLE_MAJC_OUTPUT_DROP_CACHE = "table.custom.majc.output.drop.cache";
```

Comment:
This is still a public API addition that can't occur in a patch release. If anybody references this constant, they won't be able to downgrade back to 2.1.0.

Also, based on the earlier discussion about when to use custom properties (for pluggable components) vs. built-in properties (for provided frameworks/built-in code), this does seem more suitable for a built-in property in Property.java. Although we try to avoid additions in patch releases, it's probably fine here, as long as we've considered the implications for downgraders. In this case, I don't think they are seriously harmed if they set this to true, then downgrade and it is ignored... so long as it's actually ignored and doesn't cause an error if set in 2.1.0 after downgrading.

Reply:
I moved the property back to Property.java and removed all changes from CompactionConfigurer.